src/module-elasticsuite-core/Index/IndexSettings.php

```diff
@@ -226,6 +226,74 @@ public function getDynamicIndexSettings($store)
         return $settings;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getTimeBeforeGhost(): int
+    {
+        return $this->helper->getTimeBeforeGhost();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function parseIndexName(string $indexName): array
+    {
+        // 1. Remove prefix
+        $alias = $this->helper->getIndexAlias();
+        if (!str_starts_with($indexName, "${alias}_")) {
+            throw new \LogicException(sprintf('Index %s is not an elasticsuite index.', $indexName));
+        }
+        $indexName = str_replace("${alias}_", '', $indexName);
+
+        // 2. Get date and remove suffix
+        $matches = [];
+        $date = null;
+        $isTrackingIndex = str_contains($indexName, 'tracking');
+        $pattern = $isTrackingIndex ? '{{YYYYMM}}' : $this->helper->getIndicesPattern();
+        preg_match_all('/{{([\w]*)}}/', $pattern, $matches);
+
+        if (!empty($matches[1])) {
+            $format = '';
+            foreach ($matches[1] as $value) {
+                $format .= $value;
+            }
+            $count = strlen($format);
+
+            try {
+                $date = substr(preg_replace('/[^0-9]/', '', $indexName), -$count);
+                $indexName = substr($indexName, 0, -strlen(preg_replace('/{{|}}/', '', $pattern)) - 1);
+
+                // Tracking indices are built monthly and does not fit with standard pattern containing datetime with hours.
+                $date = $isTrackingIndex ? false : new \Zend_Date($date, $format);
+            } catch (\Zend_Date_Exception $e) {
+                $date = false;
+            }
+        }
+
+        // 3. Find entity type and store code
+        $availableTypes = array_keys($this->indicesConfig->get());
+        $indexIdentifier = null;
+        $storeCode = null;
+        usort($availableTypes, function ($itemA, $itemB) {
+            return substr_count($itemB, '_') <=> substr_count($itemA, '_');
+        });
+        foreach ($availableTypes as $type) {
+            if (str_ends_with($indexName, $type)) {
+                $indexIdentifier = $type;
+                $storeCode = preg_replace("/_$type$/", '', $indexName);
+                break;
+            }
+        }
+
+        return [
+            'prefix' => $alias,
+            'store_code' => $storeCode,
+            'index_identifier' => $indexIdentifier,
+            'suffix' => $date,
```

Comment:
Since the suffix will always be the "timestamp" part, I wonder if a more specific name would not be better...